test/test_helpers.py

```diff
@@ -26,49 +26,60 @@
 
 
 def fake_script(test_case, name, content):
+    # raise unittest.SkipTest("fake_script not working on windows yet")
     if not hasattr(test_case, 'fake_script_path'):
         fake_script_path = tempfile.mkdtemp('-fake_script')
-        os.environ['PATH'] = '{}:{}'.format(fake_script_path, os.environ["PATH"])
+        test_case._old_path = os.environ["PATH"]
+        os.environ['PATH'] = os.pathsep.join([fake_script_path, os.environ["PATH"]])
 
         def cleanup():
             shutil.rmtree(fake_script_path)
-            os.environ['PATH'] = os.environ['PATH'].replace(fake_script_path + ':', '')
+            os.environ['PATH'] = test_case._old_path
 
         test_case.addCleanup(cleanup)
         test_case.fake_script_path = pathlib.Path(fake_script_path)
 
     template_args = {
         'name': name,
-        'path': test_case.fake_script_path,
+        'path': test_case.fake_script_path.as_posix(),
         'content': content,
     }
 
-    with (test_case.fake_script_path / name).open('wt') as f:
+    path = test_case.fake_script_path / name
+    with path.open('wt') as f:
         # Before executing the provided script, dump the provided arguments in calls.txt.
         # ASCII 1E is RS 'record separator', and 1C is FS 'file separator', which seem appropriate.
         f.write('''#!/bin/sh
 {{ printf {name}; printf "\\036%s" "$@"; printf "\\034"; }} >> {path}/calls.txt
 {content}'''.format_map(template_args))
-    os.chmod(str(test_case.fake_script_path / name), 0o755)
+    os.chmod(str(path), 0o755)
+    # TODO fix path maybe
```

Comment:
this got old or still valid?

Reply:
still valid, but unclear. I'll clarify.
it's hardcoding the path to bash.exe, which might not work in all installations (but works in the two i run the tests on).